### DISCUSSION TITLE
Python version

### DISCUSSION THREAD

**Discussion by hungl6844** [gekigek99/minecraft-vanilla-server-hibernation#85] [Q&A] [answered]
I know this may seem like a silly question, but as I know Python well, while Golang probably isn't in my near future, why was the Python version deprecated? Does @gekigek99 just prefer Go? Also, where can I find said deprecated Python version?

**Comment by gekigek99** [comment, marked as answer]
> why was the Python version deprecated?

It was a problem for quality of code, velocity of developement and efficiency during execution:
- not having to implement the same thing on 2 completely different languages gives me more time to make only 1 thing but better
- this rep is getting more and more features as time passes and focusing on ideas/code instead of the translation of functions between 2 languages allows me to be faster
- i love python but being a interpreted language it's not really good to implement a proxy-like program like msh

> Does @gekigek99 just prefer Go?

i would not say that... python is great to develop extremely complex things with 2 lines of code and there is a ton of documentation on how to do anything... but golang being compiled it's faster and it helps a lot in the development (if the grammar checking in vscode passes, you are sure that your variables are set up correctly.... in python you might find yourself comparing an int and a string without noticing (silly example but it's kinda true))

one thing that I miss in golang are stucts and relative methods which are a pain to implement but it's not that of a problem since msh only manages 1 server at a time... ah and sometimes python has some more complex configs more easily accessible (think about the tcp package))

>  where can I find said deprecated Python version?

you can check the deprecated python version [here](https://github.com/gekigek99/minecraft-server-hibernation/tree/v9.12) (beware that in since then a lot of things have been improved ;) )
